wip

maxhbr opened the pull request:
to test dockerized:
```
$ DOCKER_BUILDKIT=1 docker build . --tag ort/with_opossum --network=host
$ cd dir/with/scan-result.yml
$ docker run -i --rm -v /etc/group:/etc/group:ro -v /etc/passwd:/etc/passwd:ro -u 1000:1000 -v /home/mhuber/.ort/dockerHome:/home/mhuber -v $(pwd):/workdir -w /workdir --net=host ort/with_opossum:latest --force-overwrite --debug report -f Opossum -i /workdir/scan-result.yml -o /workdir
```

sschuberth committed:
Scanner: Make scanOrtResult() a top-level function

This is a preparation for not scanning all contents of an ORT result with
the same scanner.

Signed-off-by: Sebastian Schuberth <sebastian.schuberth@bosch.io>

sschuberth committed:
vulnerable-code: Migrate from Jackson to kotlinx.serialization

For portability and performance evaluation, migrate the independent
VulnerableCode client library to use kotlinx.serialization for
(de-)serialization.

If the migration turns out to be beneficial in terms of performance,
more independent parts of the code base might be migrated.

Signed-off-by: Sebastian Schuberth <sebastian.schuberth@bosch.io>

sschuberth committed:
FossIdClientReturnTypeTest: Fix the license header to be at the start

Signed-off-by: Sebastian Schuberth <sebastian.schuberth@bosch.io>

oheger-bosch committed:
DependencyNavigator: Extract an abstract test base class

The base class defines the test cases for the single functions of the
DependencyNavigator interface. Concrete sub classes have to provide
result files with the expected data in the correct format, so that the
implementation to be tested is used. That way the same test setup can
be used for different implementations, and the results can be compared
directly.

Signed-off-by: Oliver Heger <oliver.heger@bosch.io>

oheger-bosch committed:
OrtResult: Remove a withScopesResolved() call acting as safety net

Due to the way the single ORT commands currently load result files,
this call - executed when constructing the structure of projects - is
not needed, as the transformation to the dependency tree format is
manually performed.

With this call in place, however, it is not possible to test the
navigation on the new data structures, since they are always
automatically converted to the classic dependency tree format.
Therefore, remove it, as this does not have any impact on the proper
functioning of ORT commands.

Signed-off-by: Oliver Heger <oliver.heger@bosch.io>

oheger-bosch committed:
DependencyNavigator: Invoke getStableReference() where necessary

In getShortestPathsForScope() references to dependency nodes are
stored. This requires that a stable reference has to be obtained
first.

Signed-off-by: Oliver Heger <oliver.heger@bosch.io>

oheger-bosch committed:
DependencyNavigator: Add an implementation for dependency graphs

This implementation allows accessing and traversing over the
dependency references contained in a DependencyGraph object. It can
deal with graphs storing their data as DependencyReference objects
and with graphs based on lists with nodes and edges.

Signed-off-by: Oliver Heger <oliver.heger@bosch.io>

oheger-bosch committed:
Add CompatibilityDependencyNavigator class

This is a DependencyNavigator implementation that can handle the
dependency information in all kinds of OrtResults, no matter whether
it is stored in the classic tree, in the graph, or a mixed format.
This ensures compatibility with older result files.

Signed-off-by: Oliver Heger <oliver.heger@bosch.io>

oheger-bosch committed:
OrtResult: Create the most efficient dependency navigator

Use CompatibilityDependencyNavigator.create() to create the dependency
navigator for this result. This makes sure that the navigator fits
best to the dependency information stored in this result.

Signed-off-by: Oliver Heger <oliver.heger@bosch.io>

oheger-bosch committed:
AnalyzerResultBuilder: Create a dependency graph on building a result

In future, no conversion of the dependency representation is going to
be done on deserialization of ORT results, as the DependencyNavigator
of the result can handle all cases. It is, therefore, also not
necessary to do a conversion to the dependency graph format each time
a result is written.

So, do the conversion to the dependency graph format once when the
result is initially created. This is a more straight-forward mechanism
than using a custom serializer. Another advantage is that it does not
cause unexpected changes on old result files when they are read and
written out again.

As the projects created by the analyzer now use the dependency graph
format per default, also change the dummy project created by
PackageManager in case an exception occurs in resolveDependencies().

Adapt a few expected test results slightly, because empty scopes are
no longer output.

Signed-off-by: Oliver Heger <oliver.heger@bosch.io>

oheger-bosch committed:
cli: Remove the withResolvedScopes() call when loading an OrtResult

It is no longer necessary to convert the dependency information in the
result to the dependency tree format; the DependencyNavigator provided
by the result makes the concrete representation transparent.

Resolves #4111.

Signed-off-by: Oliver Heger <oliver.heger@bosch.io>

oheger-bosch committed:
AnalyzerResult: Drop the custom serializer

As no conversion of dependency information takes place any more when
loading an OrtResult, there is also no need for an automatic
conversion on saving.

AnalyzerResult objects are created in the dependency graph format by
AnalyzerResultBuilder and then remain constant during the single ORT
processing steps. Old result files are also no longer converted
automatically when they are loaded and saved again.

Signed-off-by: Oliver Heger <oliver.heger@bosch.io>

nnobelis committed:
fossid-webapp: Log pending files count

Signed-off-by: Nicolas Nobelis <nicolas.nobelis@bosch.io>

sschuberth committed:
Dockerfile: Use a sub-shell when changing the working directory

While despite the docs at [1] this does not silence the hadolint
warning, see [2], it is still considered good practice to do so.

Also fix the indentation while at it.

[1]: https://github.com/hadolint/hadolint/wiki/DL3003
[2]: https://github.com/hadolint/hadolint/issues/422

Signed-off-by: Sebastian Schuberth <sebastian.schuberth@bosch.io>

sschuberth committed:
Dockerfile: Follow the hadolint recommendation to not cache Python packages

See https://github.com/hadolint/hadolint/wiki/DL3042.

Signed-off-by: Sebastian Schuberth <sebastian.schuberth@bosch.io>

nnobelis committed:
fossid-webapp: Rename ScanStatus to ScanDescription

Signed-off-by: Nicolas Nobelis <nicolas.nobelis@bosch.io>

nnobelis committed:
fossid-webapp: Add new properties to FossID responses

This is a change for FossID 21.2.

Signed-off-by: Nicolas Nobelis <nicolas.nobelis@bosch.io>

nnobelis committed:
fossid-webapp: Remove unecessary logging

Signed-off-by: Nicolas Nobelis <nicolas.nobelis@bosch.io>

nnobelis committed:
fossid-webapp: Rename ScanState to ScanStatus

Signed-off-by: Nicolas Nobelis <nicolas.nobelis@bosch.io>

nnobelis committed:
fossid-webapp: Add new Scan Statuses

This is a change for FossID 21.2.

Signed-off-by: Nicolas Nobelis <nicolas.nobelis@bosch.io>

nnobelis committed:
fossid-webapp: Also trigger the scan when it is in NEW status

 This is a change for FossID 21.2.

Signed-off-by: Nicolas Nobelis <nicolas.nobelis@bosch.io>

maxhbr committed:
Opossum: Add initial working implementation

Signed-off-by: Maximilian Huber <maximilian.huber@tngtech.com>

oheger-bosch committed:
fossid-webapp: Add a unit test for FossId

The class contains non-trivial logic and therefore should be tested.
The new test should cover the major part of the functionality
available.

Minor changes have been applied to FossId to simplify testing. Also,
the visibility of Scanner.scanPackages() has been increased to
"internal", so that it can be called from the test class.

Signed-off-by: Oliver Heger <oliver.heger@bosch.io>

nnobelis committed:
fossid-webapp: Add test for fetching the version in FossID 21.2

Signed-off-by: Nicolas Nobelis <nicolas.nobelis@bosch.io>

nnobelis committed:
fossid-webapp: Change the version extracted from FossID

Also remove the comment saying that a dedicated function will come in
FossID 21.2: this is not the case.

Signed-off-by: Nicolas Nobelis <nicolas.nobelis@bosch.io>

nnobelis committed:
fossid-webapp: Create the service based on FossID version

Signed-off-by: Nicolas Nobelis <nicolas.nobelis@bosch.io>

sschuberth committed:
Gradle: Upgrade to Kotlin 1.5.30

Signed-off-by: Sebastian Schuberth <sebastian.schuberth@bosch.io>

sschuberth committed:
Gradle: Correct a comment about the use of the toolchain mechanism

Signed-off-by: Sebastian Schuberth <sebastian.schuberth@bosch.io>

oheger-bosch committed:
DependencyGraph: Further optimize the storage structures

The current data structures based on trees of DependencyReference
objects have the problem that the references pointing to identical
subtrees get duplicated when a graph is serialized and deserialized
again. To prevent this, store the graph as lists of nodes and edges,
which should in most cases be even more compact. This information is
sufficient to navigate the graph in the same way as before.

Keep the scopeRoots field to remain backwards compatible with existing
OrtResult files. In later changes, DependencyGraphBuilder will be
adapted to use the new structures.

Signed-off-by: Oliver Heger <oliver.heger@bosch.io>

oheger-bosch committed:
DependencyGraphBuilder: Use structures with nodes and edges

Convert the trees with DependencyReferences to lists of nodes and
edges and construct the DependencyGraph with this information.

Signed-off-by: Oliver Heger <oliver.heger@bosch.io>

nnobelis committed:
fossid-webapp: Make checkScanStatus version aware

This is a change for FossID 21.2.

Signed-off-by: Nicolas Nobelis <nicolas.nobelis@bosch.io>

sschuberth committed:
Conan: Take local configuration (like remotes) into account

This applies configuration stored to a conventional "conan_config"
directory before the dependency resolution. This is needed in order to
support custom remotes besides the default ConanCenter.

Resolves #4312.

Signed-off-by: Sebastian Schuberth <sebastian.schuberth@bosch.io>

oheger-bosch committed:
DependencyGraph: Improve documentation

Describe the single components of the representation of the dependency
graph, which also appear in the serialized form in ORT result files.

Signed-off-by: Oliver Heger <oliver.heger@bosch.io>

sschuberth committed:
SpdxDocumentFile: Fix a typo in getSpdxPackageForId() docs

While "an" probably was supposed to say "and", it should actually say
"or" anyway.

Signed-off-by: Sebastian Schuberth <sebastian.schuberth@bosch.io>

sschuberth committed:
SpdxDocumentFile: Slightly improve an exception message wording

Do not emphasize "single" so much.

Signed-off-by: Sebastian Schuberth <sebastian.schuberth@bosch.io>

sschuberth committed:
SpdxDocumentFile: Create issues for wrong checksums of referenced SPDX files

Signed-off-by: Sebastian Schuberth <sebastian.schuberth@bosch.io>

sschuberth committed:
SpdxDocumentFile: Introduce a variable in getSpdxPackage() for clarity

Signed-off-by: Sebastian Schuberth <sebastian.schuberth@bosch.io>

sschuberth committed:
SpdxDocumentFile: Make resolveExternalDocumentReference an extension function

Like getSpdxPackage(), make it an extension function on
SpdxExternalDocumentReference for readability. Fix the function
documentation while at it.

Signed-off-by: Sebastian Schuberth <sebastian.schuberth@bosch.io>

sschuberth committed:
SpdxDocumentFileFunTest: Fix checksums of referenced documents

Checksums were not consistent across platforms due to line ending
differences. Avoid this by normalizing line endings via gitattributes.

Signed-off-by: Sebastian Schuberth <sebastian.schuberth@bosch.io>

nnobelis committed:
Dockerfile: Update Cargo to the available upstream version

Signed-off-by: Nicolas Nobelis <nicolas.nobelis@bosch.io>

oheger-bosch committed:
fossid-webapp: Let FossIdConfig create the FossIdServiceWithVersion

One reason for the FossIdConfig class is to make the injection of mock
objects into FossId possible. This can only be achieved if the
configuration creates the final service object used by FossId.

Signed-off-by: Oliver Heger <oliver.heger@bosch.io>

sschuberth committed:
PipFunTest: Update the expected test result

Signed-off-by: Sebastian Schuberth <sebastian.schuberth@bosch.io>

sschuberth committed:
Gradle: Upgrade the detekt plugin to version 1.18.0

See https://github.com/detekt/detekt/releases/tag/v1.18.0.

Signed-off-by: Sebastian Schuberth <sebastian.schuberth@bosch.io>

fviernau committed:
Dockerfile: Upgrade Node to version 16

Upgrading Node from version 12 to version 16 also leads to an upgrade of
Npm. So, pin the Npm version accordingly to the latest release. See also
[1] and [2].

This change fixes [3] for unknown reason.

[1] https://github.com/nodejs/node/blob/7ca38f05a023666274569343f128c5aed81599f3/doc/changelogs/CHANGELOG_V16.md
[2] https://github.com/npm/cli/blob/59b9851d1649b8f78203cae187b1337f2441219d/CHANGELOG.md#v7206-2021-08-12
[3] https://stackoverflow.com/questions/68021121/fsevents-not-accessible-from-jest-haste-map.

Signed-off-by: Frank Viernau <frank.viernau@here.com>

oheger-bosch committed:
fossid-webapp: Add a config property to limit delta scans

Via the new "deltaScanLimit" property, users can control how many
delta scans are kept for a specific repository. In a follow-up commit,
FossId is going to evaluate this property and actually enforce this
limit.

Signed-off-by: Oliver Heger <oliver.heger@bosch.io>

sschuberth committed:
SpdxDocumentFile: Improve the exception message about a nonexisting file

Signed-off-by: Sebastian Schuberth <sebastian.schuberth@bosch.io>

sschuberth committed:
SpdxDocumentFile: Refer to the definitionFile in issue messages

To be able to match isue messages unambiguously, include the
definitionFile to the message. Being able to do so is achieved by
passing the definitionFile instead of its parent, the workingDir.

Signed-off-by: Sebastian Schuberth <sebastian.schuberth@bosch.io>

JakobSchubert commented on reporter/src/main/kotlin/reporters/OpossumReporter.kt:
```diff
@@ -0,0 +1,337 @@
+ * A [Reporter] that generates an [Opossum Input].
+ *
+ * This reporter supports the following options:
+ * - *output.file.formats*: The list of [FileFormat]s to generate, defaults to [FileFormat.JSON].
```
Right now, there actually are no other options than JSON and never will be, right? So this `output.file.formats` option is maybe a bit misleading.

sschuberth committed:
Gradle: Upgrade GreenMail to version 1.6.5

See [1].

[1]: https://github.com/greenmail-mail-test/greenmail/releases/tag/release-1.6.5

Signed-off-by: Sebastian Schuberth <sebastian.schuberth@bosch.io>

sschuberth committed:
Gradle: Upgrade AsciidoctorJ to version 2.5.2

See [1].

[1]: https://github.com/asciidoctor/asciidoctorj/blob/main/CHANGELOG.adoc#252-2021-08-08

Signed-off-by: Sebastian Schuberth <sebastian.schuberth@bosch.io>

sschuberth committed:
Gradle: Upgrade Exposed to version 0.33.1

See [1].

[1]: https://github.com/JetBrains/Exposed/blob/master/docs/ChangeLog.md#0331

Signed-off-by: Sebastian Schuberth <sebastian.schuberth@bosch.io>

oheger-bosch committed:
fossid-webapp: Enforce the limit of delta scans per project

After successfully creating a new delta scan for a repository, delete
older scans until the configured maximum number of delta scans is
reached.

Signed-off-by: Oliver Heger <oliver.heger@bosch.io>

JakobSchubert commented on reporter/src/main/kotlin/reporters/OpossumReporter.kt:
```diff
@@ -0,0 +1,337 @@
+            log.debug("add project ${project.id}")
+            val definitionFilePath = relRoot.resolve(project.definitionFilePath)
+            val projectId = project.id
+            // addPackageRoot(projectId, definitionFilePath)
```
Why is this commented out?

maxhbr replied:
It is not yet clear which of the two options is the better one ;)

JakobSchubert commented on reporter/src/main/kotlin/reporters/OpossumReporter.kt:
```diff
@@ -0,0 +1,337 @@
+        fun addProject(project: Project, curatedPackages: SortedSet<CuratedPackage>, relRoot: File = File("")) {
+            log.debug("add project ${project.id}")
+            val definitionFilePath = relRoot.resolve(project.definitionFilePath)
+            val projectId = project.id
```
I'm not sure if one should introduce this local variable. If you really want it, I would introduce it at the beginning of the function and also use it in the debug log.

sschuberth committed:
PipFunTest: Update expected test results

Signed-off-by: Sebastian Schuberth <sebastian.schuberth@bosch.io>

JakobSchubert commented on reporter/src/main/kotlin/reporters/OpossumReporter.kt:
```diff
@@ -0,0 +1,337 @@
+
+        fun addPackageRoot(id: Identifier, path: File) {
+            if (packageToRoot.containsKey(id)) {
+                packageToRoot.get(id)?.add(path)
```
Braucht man hier das Fragezeichen?

maxhbr replied:
Ja, die Liste hinter id ist mglw. `null`

maxhbr replied:
I could use `!!` here

JakobSchubert commented on reporter/src/main/kotlin/reporters/OpossumReporter.kt:
```diff
@@ -0,0 +1,337 @@
+                uuid.toString() to sortedMapOf(
+                    "source" to sortedMapOf(
+                        "name" to source,
+                        "documentConfidence" to 80
```
I think all existing agents set `document_confidence` to `0`, but it does not really matter: I think we never use this field anywhere.

JakobSchubert commented on reporter/src/main/kotlin/reporters/OpossumReporter.kt:
```diff
@@ -0,0 +1,337 @@
+
+        fun matchesOther(other: OpossumSignal): Boolean =
+            source == other.source
+                    && id != null && id == other.id
```
Warum ist hier das `id!=null`? Ich hätte gedacht, dass die `id` hier dem package-Teil einer hhc-Attribution entspricht. Dass die `id` null ist, heißt doch nur, dass es keine package-Information gibt? Sind zwei Opossum-Signals mit `id==null` nicht trotzdem gleich?

maxhbr replied:
nein, zwei attributionen mit `id==null` sind nicht gleich

JakobSchubert commented on reporter/src/main/kotlin/reporters/OpossumReporter.kt:
```diff
@@ -0,0 +1,337 @@
+            val tail = pathPieces.drop(1)
+
+            if (! tree.containsKey(head)) {
+                return true
```
Wir erwarten, dass dieser Code nie erreicht wird, oder? Der würde nur erreicht werden, wenn man die Methode mit einem Pfad aufruft, den es nicht gibt. Warum würde man in diesem Fall `true` zurückgeben?

maxhbr replied:
Was determined by random flip of coin

JakobSchubert commented on reporter/src/main/kotlin/reporters/OpossumReporter.kt:
```diff
@@ -0,0 +1,337 @@
+        }
+
+        fun isFile(): Boolean {
+            return tree.isEmpty()
```
Sehe ich es richtig, dass hier angenommen wird, dass es keine leeren Ordner gibt? Will man da vielleicht einen Kommentar setzen?

JakobSchubert commented on reporter/src/main/kotlin/reporters/OpossumReporter.kt:
```diff
@@ -0,0 +1,337 @@
+        fun toJson(): Map<*,*> {
+            return sortedMapOf(
+                "metadata" to sortedMapOf(
+                    "projectId" to "0",
```
Ich fände empty string für die nicht vorhandene project id etwas besser als "0". "0"suggeriert ein bisschen, dass die projectId ein integer wäre, es ist aber ja eine UUID.

maxhbr replied:
the Id is just a string and not used as anything else then a string

mnonnenmacher committed:
SetCommand: Fix loading the ORT result

To make sure that scopes are resolved properly use the helper function
that is also used in all other commands.

Signed-off-by: Martin Nonnenmacher <martin.nonnenmacher@here.com>

JakobSchubert reviewed (approved):
Sieht sehr gut aus, auch wenn mir beim Review die Datenstruktur von ORT nicht wirklich klar geworden ist. Hab ein paar kleine Anmerkungen gemacht. Es wäre natürlich schon sehr zum empfehlen, noch Unit Tests hinzuzufügen, aber ich fürchte, ich werd dich davon nicht überzeugen können :-).

sschuberth committed:
Upgrade to Gradle 7.2

See https://docs.gradle.org/7.2/release-notes.html.

Signed-off-by: Sebastian Schuberth <sebastian.schuberth@bosch.io>

sschuberth committed:
Gradle: Respect the GRADLE_USER_HOME environment variable

Signed-off-by: Sebastian Schuberth <sebastian.schuberth@bosch.io>

MarcelBochtler committed:
Readme: Fix wrong uppercase in advisor documentation

Resolves #4371.

Signed-off-by: Marcel Bochtler <marcel.bochtler@bosch.io>

sschuberth committed:
Gradle: Use the correct URL for tooling API artifacts

See [1].

[1]: https://docs.gradle.org/current/userguide/third_party_integration.html#sec:embedding_quickstart

Signed-off-by: Sebastian Schuberth <sebastian.schuberth@bosch.io>

sschuberth committed:
CODEOWNERS: Improve the comment with a link to GitHub docs

Signed-off-by: Sebastian Schuberth <sebastian.schuberth@bosch.io>

sschuberth committed:
Add @nnobelis as a code owner for the fossid-webapp client

Even if Nicolas is not a full contributor yet, he is the main author of
this client and nothing should go in unnoticed from this side.

Signed-off-by: Sebastian Schuberth <sebastian.schuberth@bosch.io>

sschuberth committed:
Analyzer: Trivially improve log output for single projects

Signed-off-by: Sebastian Schuberth <sebastian.schuberth@bosch.io>

sschuberth committed:
Sbt: Clarify a log message that not even a single project was found

Signed-off-by: Sebastian Schuberth <sebastian.schuberth@bosch.io>

oheger-bosch committed:
DependencyGraph: Fix a deserialization problem with empty graphs

acd4aa8 added the @JsonInclude(JsonInclude.Include.NON_DEFAULT)
annotation to DependencyGraph. This caused a rather aggressive
optimization when serializing an empty DependencyGraph object: the
fields for packages and scopes were dropped completely. Such an
instance could then no longer be deserialized, because these fields
are not nullable and do not define a default value.

(Note: This problem was actually encountered in practice with a DotNet
project, which produced a dependency graph for NuGet and an empty one
for DotNet.)

Fix this by assigning empty default values to the problematic fields.
This is in-line with the other fields in this class, which have a
default value, too.

Signed-off-by: Oliver Heger <oliver.heger@bosch.io>

MarcelBochtler committed:
RepositoryConfiguration: Remove chain wrapping

To be consistent within the code base.

Signed-off-by: Marcel Bochtler <marcel.bochtler@bosch.io>

MarcelBochtler committed:
CurationProvider: Create SimplePackageCurationProvider

As a curation provider that provides already specified package
curations.

Signed-off-by: Marcel Bochtler <marcel.bochtler@bosch.io>

MarcelBochtler committed:
Curations: Read package curations from the repository configuration file

Enable package curations from the project's repository configuration file
in order to enable project-specific curations.
These project-specific curations take precedence over possible
conflicting global ones, as they are more specific than global
curations.

Signed-off-by: Marcel Bochtler <marcel.bochtler@bosch.io>

oheger-bosch committed:
Vulnerability: Allow instances without references

When using the VulnerableCode advisor implementation, results have
been encountered that actually contain only an ID, but no references.
Therefore, to represent such results in ORT's vulnerability model,
remove the check for an empty list of references from the
Vulnerability class. (It is probably safer to store such results -
even if they contain limited information - than to drop them.)

Signed-off-by: Oliver Heger <oliver.heger@bosch.io>

MarcelBochtler committed:
config-file-ort-yml: Add documentation about project-specific package curations

Remove previously mentioned limitations.

Signed-off-by: Marcel Bochtler <marcel.bochtler@bosch.io>

MarcelBochtler committed:
config-file-ort-yml: Explain that the .ort.yml is used for project-specific configurations

Signed-off-by: Marcel Bochtler <marcel.bochtler@bosch.io>

sschuberth committed:
SpdxDocumentFile: Create issues instead of throwing exceptions

This way analysis is continued for remaining packages, and all potential
issues are reported at once.

Signed-off-by: Sebastian Schuberth <sebastian.schuberth@bosch.io>

oheger-bosch committed:
DependencyGraph: Collect issues from dependency nodes, too

After the introduction of dependency nodes in acd4aa8, it has been
missed to adapt collectIssues() accordingly. Fix this, so that all
issues are reported.

Signed-off-by: Oliver Heger <oliver.heger@bosch.io>

sschuberth committed:
Gradle: Append an optional "plugin" directory to the CLI runtime classpath

In order to conveniently add plugin JARs to the classpath when running
ORT, support getting these from an optional "plugin" directory. Use
singular "plugin" to resemble the also singular "lib" directory name.

Even if shortending the path on Linux is not required, still do it for
simplicity to align the code with the one for the generated Windows
script.

Signed-off-by: Sebastian Schuberth <sebastian.schuberth@bosch.io>

sschuberth committed:
Gradle: Stricten the pattern for patching the start scripts

Just to be on the safe side, to not patch a wrong location.

Signed-off-by: Sebastian Schuberth <sebastian.schuberth@bosch.io>

sschuberth committed:
Gradle: Use the correct encoding for reading start scripts

Gradle uses the platform-default encoding to write the files, so do the
same to read them.

Signed-off-by: Sebastian Schuberth <sebastian.schuberth@bosch.io>

MarcelBochtler committed:
RepositoryConfiguration: Add possibility to add package-configurations

Signed-off-by: Marcel Bochtler <marcel.bochtler@bosch.io>

sschuberth committed:
Scanner: Document scanOrtResult()'s "skipExcluded" parameter

This is a fixup for 910b9aa.

Signed-off-by: Sebastian Schuberth <sebastian.schuberth@bosch.io>

sschuberth committed:
Use named parameters for calls to requestPasswordAuthentication()

This is to get a better overview for the confusingly high number of
parameters of this function.

Signed-off-by: Sebastian Schuberth <sebastian.schuberth@bosch.io>

sschuberth committed:
Consistently use the static requestPasswordAuthentication() function

That convenience function uses the default authenticator instance to
request a server type authentication. It is already used by the Git
class.

Signed-off-by: Sebastian Schuberth <sebastian.schuberth@bosch.io>

sschuberth committed:
Git: Fix passing the URI scheme as the protocol for authentication

What a URI calls the "scheme" is what a URL calls the "protocol".
However, what an authenticator calls the "scheme" is the "authentication
scheme", not the "protocol".

Signed-off-by: Sebastian Schuberth <sebastian.schuberth@bosch.io>

sschuberth committed:
Git: Do not pass a prompt for authentication

ORT does not want to trigger any interactive authentication prompt here,
so simply do not provide any prompt at all.

Signed-off-by: Sebastian Schuberth <sebastian.schuberth@bosch.io>

sschuberth committed:
FossId: Pass on more URI properties for authentication

While these properties are barely used, pass them on for consistency.

Signed-off-by: Sebastian Schuberth <sebastian.schuberth@bosch.io>

MarcelBochtler committed:
Evaluator: Apply project-local package-configurations

Package-configurations must be unique per ID and provenance even
if one configuration is placed in the `.ort.yml` and the other one
is placed in an ORT-wide package-configurations directory.

Signed-off-by: Marcel Bochtler <marcel.bochtler@bosch.io>

MarcelBochtler committed:
Readme: Add documentation about project-local package-configurations

Signed-off-by: Marcel Bochtler <marcel.bochtler@bosch.io>

sschuberth committed:
Conan: Forward Java authenticator credentials

Conan does not support parsing .netrc for credentials to authenticate
against remotes, see [1]. So forward any credentials for remote hosts
provided by the Java authenticator, which in this case is an instance of
OrtAuthenticator that supports getting credentails from a .netrc file.

[1] https://github.com/conan-io/conan/issues/3046#issuecomment-904516572

Signed-off-by: Sebastian Schuberth <sebastian.schuberth@bosch.io>

sschuberth committed:
FossId: Improve an error log message

In particular, use lazy evaluation for the message.

Signed-off-by: Sebastian Schuberth <sebastian.schuberth@bosch.io>

sschuberth committed:
FossId: Fix issues to name the scanner as the source

By convention, the source of an issue should be the name of the tool
causing the issue. Fix this, and while at it also slightly reword the
messages for readability when mentioning the package coordinates.

Signed-off-by: Sebastian Schuberth <sebastian.schuberth@bosch.io>

sschuberth committed:
FossId: Continue the processing of other packages after a non-Git package

Signed-off-by: Sebastian Schuberth <sebastian.schuberth@bosch.io>

sschuberth committed:
Consistently capitalize the "Git" VCS type in docs / assets

Signed-off-by: Sebastian Schuberth <sebastian.schuberth@bosch.io>

sschuberth committed:
FossIdTest: Suffix function names with "()" for test name readability

Signed-off-by: Sebastian Schuberth <sebastian.schuberth@bosch.io>

MarcelBochtler committed:
config-file-ort-yml: Fix example for project-local curations

This is a fixup for f4a0ea166372635fe0bf0651dabc6928611118e4.

Signed-off-by: Marcel Bochtler <marcel.bochtler@bosch.io>

Commit:
EvaluatedModel: Add vulnerabilities

Include the vulnerabilities and vulnerabilities resolutions to
EvaluatedModel in order to prepare for showing them in the WebApp.

Signed-off-by: Artem Shykhov <ext-artem.shykhov@here.com>

Commit:
EvaluatedModel: Add tests for vulnerabilities

Add test case when advisor found vulnerable package and assert
that all data about vulnerability successfully transferred
to serialized representation of EvaluatedModel in YAML and JSON.

Signed-off-by: Artem Shykhov <ext-artem.shykhov@here.com>

sschuberth committed:
Gradle: Upgrade kotest to version 4.6.2

See https://kotest.io/docs/changelog.html#462-august-2021.

Signed-off-by: Sebastian Schuberth <sebastian.schuberth@bosch.io>

sschuberth committed:
AbstractDependencyNavigatorTest: Avoid converting a sequence to a list

This is now possible thanks to the fix for [1].

[1]: https://github.com/kotest/kotest/issues/2412

Signed-off-by: Sebastian Schuberth <sebastian.schuberth@bosch.io>

sschuberth committed:
AbstractDependencyNavigatorTest: Make a property private

Protected visibility is not needed here.

Signed-off-by: Sebastian Schuberth <sebastian.schuberth@bosch.io>

maxhbr committed:
add comments from review

Signed-off-by: Maximilian Huber <maximilian.huber@tngtech.com>

maxhbr committed:
add comments from review

Signed-off-by: Maximilian Huber <maximilian.huber@tngtech.com>

sschuberth committed:
VulnerableCodeService: Rename a create() parameter to just "url"

For consistency with other client implementations.

Signed-off-by: Sebastian Schuberth <sebastian.schuberth@bosch.io>

sschuberth committed:
FossIdConfig: Reduce code duplication when creating the service

Signed-off-by: Sebastian Schuberth <sebastian.schuberth@bosch.io>

sschuberth committed:
FossIdServiceWithVersion: Inline some variables for shortness

Signed-off-by: Sebastian Schuberth <sebastian.schuberth@bosch.io>

sschuberth committed:
fossid-webapp: Move getFossIdVersion() to the public extension functions

This is useful to have as part of the public client API.

Signed-off-by: Sebastian Schuberth <sebastian.schuberth@bosch.io>

sschuberth committed:
fossid: Improve logging of the FossID server version

Log the version only once instead for every scan, and slightly improve
the message wording.

Signed-off-by: Sebastian Schuberth <sebastian.schuberth@bosch.io>

sschuberth committed:
FossId: Unify issue logging via createAndLogIssue()

Signed-off-by: Sebastian Schuberth <sebastian.schuberth@bosch.io>

sschuberth committed:
FossId: Quote the scan code in logs

For consistent readability.

Signed-off-by: Sebastian Schuberth <sebastian.schuberth@bosch.io>

sschuberth committed:
PipFunTest: Update expected test output

Signed-off-by: Sebastian Schuberth <sebastian.schuberth@bosch.io>

maxhbr committed:
recurse dependencies

Signed-off-by: Maximilian Huber <maximilian.huber@tngtech.com>

sschuberth committed:
FossId: Skip packages with an empty VCS revision

RepositoryProvenance fails hard when passed an empty revision. Avoid
that and continue with remaining packages by skipping packages with an
empty VCS revision.

Signed-off-by: Sebastian Schuberth <sebastian.schuberth@bosch.io>

maxhbr committed:
add license information and generate gz

Signed-off-by: Maximilian Huber <maximilian.huber@tngtech.com>

maxhbr committed:
add license information and generate gz

Signed-off-by: Maximilian Huber <maximilian.huber@tngtech.com>

maxhbr committed:
more clean / verbose logging

Signed-off-by: Maximilian Huber <maximilian.huber@tngtech.com>

sschuberth committed:
PipFunTest: Update expected test output

Signed-off-by: Sebastian Schuberth <sebastian.schuberth@bosch.io>

sschuberth committed:
FossId: Slightly shorten a log message for compactness

Signed-off-by: Sebastian Schuberth <sebastian.schuberth@bosch.io>

sschuberth committed:
FossId: Fix wording of a log message

Signed-off-by: Sebastian Schuberth <sebastian.schuberth@bosch.io>

sschuberth committed:
FossId: Make "existingScanCode" variables use camel-case

This is for consistency with "scanCode" variables only.

Signed-off-by: Sebastian Schuberth <sebastian.schuberth@bosch.io>

sschuberth committed:
FossId: Rework and align messages that mention the scan code

Signed-off-by: Sebastian Schuberth <sebastian.schuberth@bosch.io>

sschuberth committed:
fossid-webapp: Consistently refer to new FossID version as 2021.2

This makes more clear that CalVer is used.

Signed-off-by: Sebastian Schuberth <sebastian.schuberth@bosch.io>

MarcelBochtler committed:
config-file-ort-yml: Fix wrong documentation about project-local package-configurations

This is a fixup for e2b6a34e0a029308f63daf75dda4e1b42a846cb0.

Signed-off-by: Marcel Bochtler <marcel.bochtler@bosch.io>

maxhbr committed:
more clean / verbose logging

Signed-off-by: Maximilian Huber <maximilian.huber@tngtech.com>

maxhbr committed:
more clean / verbose logging

Signed-off-by: Maximilian Huber <maximilian.huber@tngtech.com>

maxhbr committed:
add tests

Signed-off-by: Maximilian Huber <maximilian.huber@tngtech.com>

maxhbr commented:
replaced by https://github.com/maxhbr/oss-review-toolkit/pull/10, heading to master
